packages/espree/lib/options.js

```diff
@@ -22,19 +26,30 @@ const SUPPORTED_VERSIONS = [
     15, // 2024
     16, // 2025
     17 // 2026
-];
+]);
+
+/**
+ * @typedef {typeof SUPPORTED_VERSIONS[number]} normalizedEcmaVersion
+ */
+
+const LATEST_ECMA_VERSION =
+    /* eslint-disable jsdoc/valid-types -- Bug */
+    /** @type {typeof SUPPORTED_VERSIONS extends readonly [...infer _R, infer L] ? L : never} */ (
+        SUPPORTED_VERSIONS.at(-1)
+        /* eslint-enable jsdoc/valid-types -- Bug */
+    );
 
 /**
  * Get the latest ECMAScript version supported by Espree.
- * @returns {number} The latest ECMAScript version.
+ * @returns {typeof LATEST_ECMA_VERSION} The latest ECMAScript version.
  */
 export function getLatestEcmaVersion() {
-    return SUPPORTED_VERSIONS.at(-1);
+    return LATEST_ECMA_VERSION;
 }
 
 /**
  * Get the list of ECMAScript versions supported by Espree.
- * @returns {number[]} An array containing the supported ECMAScript versions.
+ * @returns {normalizedEcmaVersion[]} An array containing the supported ECMAScript versions.
```

Comment:
```suggestion
 * @returns {[...typeof SUPPORTED_VERSIONS]} An array containing the supported ECMAScript versions.
```

Like in the comment https://github.com/eslint/js/pull/705#discussion_r2643219478, I think it would be nice to narrow the return type of `getSupportedEcmaVersions` slightly.

Currently, the return type of `getSupportedEcmaVersions()` is typed as follows:

<img width="688" height="63" alt="Image" src="https://github.com/user-attachments/assets/64ed3534-b795-41a1-8485-f18c5481d251" />

```ts
(3 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17)[]
```

This means an array like `[3, 3, 3, 3, 3, 3, 3]` can satisfy `(3 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17)[]`.

However, if we use `[...typeof SUPPORTED_VERSIONS]` as the return type, the type can be expressed more precisely, like below:

<img width="586" height="71" alt="Image" src="https://github.com/user-attachments/assets/98262b30-cca8-4d0c-b40d-48e0c04c0d31" />

```ts
[3, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17]
```

Since the `getSupportedEcmaVersions()` function returns the `SUPPORTED_VERSIONS` array items unchanged, I think that would work. What do you think?